diff -ruNa old/pith/smkeys.c new/pith/smkeys.c
--- old/pith/smkeys.c	2010-07-09 19:27:11.000000000 +0200
+++ new/pith/smkeys.c	2010-07-09 19:36:42.000000000 +0200
@@ -277,11 +277,26 @@
 get_x509_subject_email(X509 *x)
 {
     char *result = NULL;
-    STACK *emails = X509_get1_email(x);
+#if OPENSSL_VERSION_NUMBER >= 0x1000000f /* OpenSSL 1.0.0 */
+    STACK_OF(OPENSSL_STRING) *emails;
+#else /* OpenSSL 0.x and 1.0.0-dev/beta */
+    STACK *emails;
+#endif
+
+    emails = X509_get1_email(x);
+
+#if OPENSSL_VERSION_NUMBER >= 0x1000000f /* OpenSSL 1.0.0 */
+    if (sk_OPENSSL_STRING_num(emails) > 0) {
+	/* take the first one on the stack */
+	result = cpystr(sk_OPENSSL_STRING_value(emails, 0));
+    }
+#else /* OpenSSL 0.x and 1.0.0-dev/beta */
     if (sk_num(emails) > 0) {
 	/* take the first one on the stack */
 	result = cpystr(sk_value(emails, 0));
     }
+#endif
+
     X509_email_free(emails);
     return result;
 }
